schemas/post-artifact-request.yml

```diff
@@ -3,6 +3,110 @@ title:              "Post Artifact Request"
 description: "Request a authorization to put and artifact or posting of a URL as an artifact. Note that the `storageType` property is referenced in the response as well."
 type:               object
 oneOf:
+  - title: "Blob Artifact Request"
+    description: |
+      Request a list of requests in a generalized format which can be run to
+      upload an artifact to storage managed by the queue.
+    type: object
+    properties:
+      storageType:
+        description: Artifact storage type, in this case `'blob'`
+        type: string
+        enum:
+          - blob
+      expires:
+        description: |
+          Date-time after which the artifact should be deleted. Note, that
+          these will be collected over time, and artifacts may remain
+          available after expiration. S3 based artifacts are identified in
+          azure table storage and explicitly deleted on S3 after expiration.
+        type: string
+        format: date-time
+      contentType:
+        description: |
+          Artifact mime-type, when uploading artifact to the signed
+          `PUT` URL returned from this request this must given with the
+           `ContentType` header. Please, provide correct mime-type,
+           this make tooling a lot easier, specifically,
+           always using `application/json` for JSON artifacts.
+        type: string
+        maxLength: 255
+      encoding:
+        description: |
+          Optionally provide an encoding type which should be set as the HTTP
+          Content-Encoding header for this artifact.
+        type: string
+        maxLength: 255
+      sha256:
+        description: |
+          The complete SHA256 value of the entire artifact.  This must be the
+          SHA256 of the file which is to be uploaded.  For single part uploads,
+          the upload will fail if the SHA256 value of what is uploaded does not
+          match this value
+        type: string
+        pattern: "[a-fA-F0-9]{64}"
+        minLength: 64
+        maxLength: 64
+      size:
+        description: |
+          The number of bytes of the entire artifact.  This must be the number
+          of bytes in the file to be uploaded.  For single part uploads, the
+          upload will fail if the number of bytes uploaded does not match this
+          value.  A single part upload (e.g. no parts list) may be at most 5GB.
+          This limit is enforced in the code because it is not possible to
+          represent all of the restrictions in a json-schema.  A multipart
+          upload may be at most 5TB, with each part other than the last being
+          between 5MB and 5GB in size.
+        type: number
+        multipleOf: 1.0
```

Comment:
No need, use `type: integer` instead... It's a bit weird that JSON schema supports both `number` and `integer` types, but when you have to do code-gen it's really awesome to have `type: integer`.